src/languages/ruby.js

```diff
@@ -104,10 +104,25 @@ export default function(hljs) {
         begin: /%[qQwWx]?\|/,
         end: /\|/
       },
+      // in the following expressions, \B in the beginning suppresses recognition of ?-sequences
+      // where ? is the last character of a preceding identifier, as in: `func?4`
       {
-        // \B in the beginning suppresses recognition of ?-sequences where ?
-        // is the last character of a preceding identifier, as in: `func?4`
-        begin: /\B\?(\\\d{1,3}|\\x[A-Fa-f0-9]{1,2}|\\u[A-Fa-f0-9]{4}|\\?\S)\b/
+        begin: /\B\?(\\\d{1,3})/
+      },
+      {
+        begin: /\B\?(\\x[A-Fa-f0-9]{1,2})/
+      },
+      {
+        begin: /\B\?(\\u\{?[A-Fa-f0-9]{1,6}\}?)/
```

Comment:
Where does the 1-6 come from here?  Should it not be 4?  What other cases are we trying to cover? Your tests only mention "Unicode character, where nnnn is exactly 4 hexadecimal digits "

Reply:
```
\unnnn         Unicode character, where nnnn is exactly 4 hexadecimal digits ([0-9a-fA-F])
\u{nnnn ...}   Unicode character(s), where each nnnn is 1-6 hexadecimal digits ([0-9a-fA-F])
```
This regexp is accounting for these two cases.

Reply:
Ah, lets please add a hew test cases for this variant since the everything else is covered so well.

Reply:
Added the required test cases.